### PR TITLE
Add an option to enable virtual space

### LINKED ISSUE #11443: Add an option for enabling Virtual Space
### Description of the Issue
Scintilla supports Virtual Space, but there's no option for enabling it.

### Steps to Reproduce the Issue
Launch Notepad++. Be confused while working with it if you are used to Virtual Space. Unsuccessfully search for an option to free the caret from the tyranny of present characters. End up using some hacky solution.

### Expected Behavior
Clicking a nice "Enable Virtual Space" checkbox in General -> Editing.

### Actual Behavior
There's no "Enable Virtual Space" checkbox in General -> Editing.

##

Opening this issue as the prerequisite for an incoming pull request :-)

### LINKED ISSUE #11444: Add an option to prevent the arrow keys from moving cursor to previous/next line
### Description of the Issue
Caret moves to the previous or next line when moving left or right and reaching the start or the end of the lines. There's no easy way to change the behavior unless resorting to hacky methods.

### Steps to Reproduce the Issue

Load or type some lines of text. Press left until reaching the start of a line, or right until reaching the end of a line. If there's any line before or after the current one the caret will wrap to the end of the previous line or the start of the next one. Open the settings menu and go to Preferences -> Editing. No option there.

### Expected Behavior

Having a checkbox for this.

### Actual Behavior

There's no checkbox for this.

###

Opening this issue as a prerequisite for an incoming pull request.

### PR DESCRIPTION
**Pull request by m6502**
Added an option in preferences to enable or disable virtual space.

Fixes #11443 
Fixes #11444

### CODE REVIEW

**Comment by m6502** [comment]
This comes from https://github.com/notepad-plus-plus/notepad-plus-plus/issues/11443 and https://github.com/notepad-plus-plus/notepad-plus-plus/issues/11444

**Comment by dfs-** [comment]
The PR looks good to me. Would you please be so kind and add your new checkbox in the translation files at `PowerEditor\installer\nativeLang`. It will be enough to add it in `english.xml`. If you know other translations feel free to add them as well (your gif shows spanish.xml). 👍

**Comment by dfs-** [comment]
Ah, okay. Now I noticed.

You basically defined it's id in `PowerEditor/src/WinControls/Preference/preference_rc.h`. However, you put your new define between already existing ones, moving those up a number:

https://github.com/notepad-plus-plus/notepad-plus-plus/blob/20a4ee23e699a0c182bb5381d15bfeb719e9daf3/PowerEditor/src/WinControls/Preference/preference_rc.h#L126-L136

This must be changed, as the defined IDs are the ids used in the translation. Please move your `IDC_CHECK_VIRTUALSPACE` below `IDC_DISTRACTIONFREEVAL_STATIC`  giving those changed their previous number back and then defining VIRTUALSPACE as "IDD_PREFERENCE_SUB_EDITING + 45", which then makes your translation ID "6245".

File: PowerEditor/src/WinControls/Preference/preference_rc.h [hunk 1]
```diff
@@ -123,16 +123,17 @@
...
+	#define	IDC_CHECK_SCROLLBEYONDLASTLINE (IDD_PREFERENCE_SUB_EDITING + 37)
+
+	#define	IDC_STATIC_MULTILNMODE_TIP (IDD_PREFERENCE_SUB_EDITING + 38)
+	#define	IDC_COLUMNPOS_EDIT (IDD_PREFERENCE_SUB_EDITING + 39)
+	#define	IDC_CHECK_RIGHTCLICKKEEPSSELECTION (IDD_PREFERENCE_SUB_EDITING + 40)
+	#define	IDC_PADDINGLEFT_SLIDER (IDD_PREFERENCE_SUB_EDITING + 41)
+	#define	IDC_PADDINGRIGHT_SLIDER (IDD_PREFERENCE_SUB_EDITING + 42)
+	#define	IDC_PADDINGLEFTVAL_STATIC (IDD_PREFERENCE_SUB_EDITING + 43)
+	#define	IDC_PADDINGRIGHTVAL_STATIC (IDD_PREFERENCE_SUB_EDITING + 44)
+	#define	IDC_DISTRACTIONFREEVAL_STATIC (IDD_PREFERENCE_SUB_EDITING + 45)
```
Comment (dfs-) [inline PowerEditor/src/WinControls/Preference/preference_rc.h:137]: Please revert the changed IDs as it breaks translation. IDC_CHECK_VIRTUALSPACE should then be:
```
#define	IDC_CHECK_VIRTUALSPACE (IDD_PREFERENCE_SUB_EDITING + 45)
```

**Comment by m6502** [comment]
OK that explains everything, thanks a lot for your comments. I will fix it.

**Comment by m6502** [comment]
OK, here it is.

File: PowerEditor/src/WinControls/Preference/preference_rc.h (hunk 1 above)
Comment (m6502) [reply PowerEditor/src/WinControls/Preference/preference_rc.h:138]: Done

File: PowerEditor/src/Parameters.cpp [hunk 2]
```diff
@@ -6637,7 +6648,7 @@ void NppParameters::createXmlTreeFromGUIParams()
 		GUIConfigElement->SetAttribute(TEXT("customColorLinkText"), _nppGUI._darkmode._customColors.linkText);
 	}
 
-	// <GUIConfig name="ScintillaPrimaryView" lineNumberMargin="show" bookMarkMargin="show" indentGuideLine="show" folderMarkStyle="box" lineWrapMethod="aligned" currentLineHilitingShow="show" scrollBeyondLastLine="no" rightClickKeepsSelection="no" disableAdvancedScrolling="no" wrapSymbolShow="hide" Wrap="no" borderEdge="yes" edge="no" edgeNbColumn="80" zoom="0" zoom2="0" whiteSpaceShow="hide" eolShow="hide" borderWidth="2" smoothFont="no" />
+	// <GUIConfig name="ScintillaPrimaryView" lineNumberMargin="show" bookMarkMargin="show" indentGuideLine="show" folderMarkStyle="box" lineWrapMethod="aligned" currentLineHilitingShow="show" virtualSpace="no" scrollBeyondLastLine="no" rightClickKeepsSelection="no" disableAdvancedScrolling="no" wrapSymbolShow="hide" Wrap="no" borderEdge="yes" edge="no" edgeNbColumn="80" zoom="0" zoom2="0" whiteSpaceShow="hide" eolShow="hide" borderWidth="2" smoothFont="no" />
```
Comment (donho) [inline PowerEditor/src/Parameters.cpp:6651]: It's not necessary.
Please restore it.

File: PowerEditor/src/resource.h [hunk 3]
```diff
@@ -615,24 +615,25 @@
...
+	#define NPPM_INTERNAL_UPDATECLICKABLELINKS          (NOTEPADPLUS_USER_INTERNAL + 53)
+	#define NPPM_INTERNAL_SCINTILLAFINDERWRAP           (NOTEPADPLUS_USER_INTERNAL + 54)
+	#define NPPM_INTERNAL_MINIMIZED_TRAY                (NOTEPADPLUS_USER_INTERNAL + 55)
+	#define NPPM_INTERNAL_SCINTILLAFINDERCOPYVERBATIM   (NOTEPADPLUS_USER_INTERNAL + 56)
+	#define NPPM_INTERNAL_FINDINPROJECTS                (NOTEPADPLUS_USER_INTERNAL + 57)
+	#define NPPM_INTERNAL_SCINTILLAFINDERPURGE          (NOTEPADPLUS_USER_INTERNAL + 58)
+	#define NPPM_INTERNAL_UPDATETEXTZONEPADDING         (NOTEPADPLUS_USER_INTERNAL + 59)
+	#define NPPM_INTERNAL_REFRESHDARKMODE				(NOTEPADPLUS_USER_INTERNAL + 60)
+	#define NPPM_INTERNAL_SCINTILLAFINDERCOPYPATHS      (NOTEPADPLUS_USER_INTERNAL + 61)
+	#define NPPM_INTERNAL_REFRESHWORKDIR                (NOTEPADPLUS_USER_INTERNAL + 62)
```
Comment (donho) [inline PowerEditor/src/resource.h:636]: The diff should show only the above line, not the others.
Restore them please.

**Review by donho — CHANGES_REQUESTED** [review]
Please consider the requested changes.

**Comment by m6502** [comment]
I will review the code and commit it, thanks.

File: PowerEditor/src/resource.h (hunk 3 above)
Comment (m6502) [reply PowerEditor/src/resource.h]: Done.

**Comment by m6502** [comment]
Everything should be in line with the style guide now, thanks for taking a look at the PR.

**Review by donho — APPROVED** [review]

**Comment by donho** [comment]
It's OK for this time, but please create always a new branch for a new PR:
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#guidelines-for-pull-requests

**Pull request closed without merging**